crates/argmin/src/core/state/mod.rs

```diff
@@ -90,6 +90,9 @@ pub trait State {
     /// Returns current cost function evaluation count
     fn get_func_counts(&self) -> &HashMap<String, u64>;
 
+    /// Set function counting status
+    fn set_counting(self, mode: bool) -> Self;
+
```

Comment:
I don't like this being part of the `State` trait, as it leaks too much internal details in a public interface. I'd have to have a more detailed look though in order to be able to make suggestion.
Also, setters are usually not called `set_*` in argmin, therefore I'd prefer `counting`.